module/zfs/vdev.c

```diff
@@ -2811,8 +2848,19 @@ vdev_stat_update(zio_t *zio, uint64_t psize)
 				vs->vs_self_healed += psize;
 		}
 
-		vs->vs_ops[type]++;
-		vs->vs_bytes[type] += psize;
+		/*
+		 * The bytes/ops/histograms are recorded at the leaf level and
+		 * aggregated into the higher level vdevs in vdev_get_stats().
+		 */
+		if (vd->vdev_ops->vdev_op_leaf) {
+			vs->vs_ops[type]++;
+			vs->vs_bytes[type] += psize;
+			vs->vs_queue_histo[zio->io_priority]
+						[HISTO(zio->io_delta)]++;
```

Comment:
The io_delta time I believe includes the disk and queue time.  It's updated in `vdev_queue_io_done()`.